### PR TITLE
Updated CI workflow + added test report step

### PR DESCRIPTION
**Pull request by mbien**
Updated CI workflow 
- fixed some tests
- added junit report step
- it should now build/test on JDK 21

### CODE REVIEW

**Comment by mbien** [comment]
thats cool. test annotations are working here. Going to take a look at the failed test, looks like it expects a certain junit version in the local repo.

**Comment by cstamas** [comment]
+1

**Comment by cstamas** [comment]
This is what I have as well locally on Java21, it fails, and this is the relevant part of the log: https://gist.github.com/cstamas/9a2fcf938da2c2ccdf12296139643f4c

File: nbm-maven-plugin/src/it/full/pom.xml [hunk 1]
```diff
@@ -63,6 +62,14 @@ under the License.
                         </archive>
                     </configuration>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>3.2.1</version>
+                    <configuration>
+                        <argLine>-Djava.security.manager=allow --add-opens=java.base/java.net=ALL-UNNAMED</argLine>
```
Comment (cstamas) [inline nbm-maven-plugin/src/it/full/pom.xml:70]: This should go into a profile that is activated like:
```
    <activation>
      <jdk>21</jdk>
    </activation>
```

**Comment by mbien** [comment]
now its green on JDK 21 but fails everywhere else -> we are getting closer :)

File: nbm-maven-plugin/src/it/full/pom.xml (hunk 1 above)
Comment (mbien) [reply nbm-maven-plugin/src/it/full/pom.xml:70]: i hoped we could turn JDK 8 off but it fails on JDK 11 too, so yeah it would have to be set dynamically

Comment (mbien) [reply nbm-maven-plugin/src/it/full/pom.xml]: @cstamas would you like to check if I implemented the profile correctly?

**Comment by cstamas** [comment]
+1

File: nbm-maven-plugin/src/it/full/pom.xml (hunk 1 above)
Comment (cstamas) [reply nbm-maven-plugin/src/it/full/pom.xml]: lgtm

**Pull request merged**